### PR TITLE
chore(core): move plugin utility functions into a common module

### PR DESCRIPTION
**Pull request by mersinvald**

### CODE REVIEW

File: src/plugin_runtime/util.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+use crate::config::{Config, PluginDefinitionMap};
+use crate::plugin_runtime::resolver::PluginResolver;
+use crate::plugin_runtime::starter::PluginStarter;
+use crate::plugin_runtime::Injection;
+use crate::plugin_support::{Plugin, RawPlugin, RawPluginState};
+
+pub fn load_plugins_for_config(
+    config: &Config,
+    injections: Option<Vec<Injection>>,
```
Comment (AnthonyMikh) [inline src/plugin_runtime/util.rs]: Does it make sense to pass an optional `Vec`, especially considering that empty `Vec` doesn't allocate?

File: src/plugin_runtime/util.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    // Starting stage
+    let plugins = start_plugins(plugins)?;
+    log::debug!("all plugins started");
+
+    // Prepend injected plugins to plugin list
+    let plugins = injections
+        .into_iter()
+        .map(|(plugin, _)| plugin)
+        .chain(plugins.into_iter())
+        .collect();
```
Comment (AnthonyMikh) [inline src/plugin_runtime/util.rs]: This one allocates fresh new `Vec` for no good reason. This one reuses allocation if possible _and_ better reflects the intent:
`plugins.splice(..0, injections.into_iter().map(|(plugin, _)| plugin));`

Comment (mersinvald) [reply src/plugin_runtime/util.rs]: Done: https://github.com/semanteecore/semanteecore/pull/69/files#diff-4810b435c94523ae50b9c1e2c602bf8aR31-R32

File: src/plugin_runtime/util.rs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    log::debug!("all plugins started");
+
+    // Prepend injected plugins to plugin list
+    let plugins_to_inject = injections.into_iter().map(|(plugin, _)| plugin);
+    plugins.splice(..0, plugins_to_inject);
+
+    Ok(plugins)
+}
+
+fn plugin_def_map_to_vec(plugins: PluginDefinitionMap) -> Vec<RawPlugin> {
```
Comment (AnthonyMikh) [inline src/plugin_runtime/util.rs]: Is this function used in more than one place? If so, it makes sense to make it accept map by refrerence

Comment (mersinvald) [reply src/plugin_runtime/util.rs]: No, there is a consuming "fn(self)"iterator.

Comment (AnthonyMikh) [reply src/plugin_runtime/util.rs]: Sure, but in a code above it is called on `.clone`d `HashMap`, so the code clones the entire map just in order to traverse it and make in `Vec`. This is an allocation which can be avoided by passing a reference to map and cloning only necessary bits.

**Pull request merged**